app/src/main/java/waiter/Connectable/ClientConnection.java

```diff
@@ -20,11 +21,16 @@ public ClientConnection(Socket socket) throws IOException {
     }
 
     public String read() throws IOException {
-        return this.bufferedReader.readLine();
+        StringBuilder requestContents = new StringBuilder();
+        while(!requestContents.toString().contains("\r\n")) {
+            requestContents.append((char) this.bufferedReader.read());
+        }
+
+        return requestContents.toString().strip();
     }
 
     public void write(String toClient) throws IOException {
-        this.printStream.print(toClient);
+        this.printStream.write(toClient.getBytes(StandardCharsets.UTF_8));
```

Comment:
Possible breaking up of all the `.`s?

Reply:
Addressed in [this commit here](https://github.com/jonmanahan/waiter/pull/15/commits/da4cec5984fc9f2abea5b44b73e92d5d7778396c), used an intermittent variable `bytesToWriteToClient = toClient.getBytes(StandardCharsets.UTF_8)`